File: src/mcp_agent/llm/provider_types.py
"""
Type definitions for LLM providers.
"""

from enum import Enum


class Provider(Enum):
    """Supported LLM providers"""

    ANTHROPIC = "anthropic"
    OPENAI = "openai"
    FAST_AGENT = "fast-agent"
    GOOGLE = "google"
    DEEPSEEK = "deepseek"
    GENERIC = "generic"
    OPENROUTER = "openrouter"

File: src/llm/providerTypes.ts
export enum Provider {
  FAST_AGENT = 'fast_agent',
  OPENAI = 'openai',
  ANTHROPIC = 'anthropic',
  GENERIC = 'generic',
}

export enum ReasoningEffort {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}

export interface ModelConfig {
  provider: Provider;
  modelName: string;
  reasoningEffort?: ReasoningEffort;
}
